### PR TITLE
Licensing/installer

### PR DESCRIPTION
**Pull request by jimenez**
## High-level description
We need a few changes in the templates and for conf generation in order to add a new component in enterprise.

Please see description in corresponding Enterprise PR: https://github.com/mesosphere/dcos-enterprise/pull/1982 

## Corresponding DC/OS tickets (obligatory)
See ticket linked in Enterprise PR

## Checklist for all PRs
  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: Please see tests in Enterprise PR
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by jimenez** [comment]
@mesosphere-mergebot bump-ee

**Comment by jimenez** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

File: gen/calc.py [hunk 1]
```diff
@@ -1009,7 +1010,8 @@ def calculate_fault_domain_detect_contents(fault_domain_detect_filename):
         'mesos_master_work_dir': '/var/lib/dcos/mesos/master',
         'mesos_agent_work_dir': '/var/lib/mesos/slave',
         'fault_domain_detect_filename': 'genconf/fault-domain-detect',
-        'fault_domain_detect_contents': calculate_fault_domain_detect_contents
+        'fault_domain_detect_contents': calculate_fault_domain_detect_contents,
+        'license_key_contents': '',
```
Comment (klueska) [inline gen/calc.py]: Will this allow us to override this in config.yaml *instead* of having a license.txt file under `genconf`?

Comment (jimenez) [reply gen/calc.py]: cc @branden

Comment (branden) [reply gen/calc.py]: I explained how enterprise handles the license key param here: https://github.com/mesosphere/dcos-enterprise/pull/1982#discussion_r160215989

Comment (jimenez) [reply gen/calc.py]: Thanks @branden !

**Comment by jimenez** [comment]
@mesosphere-mergebot ship_it_label_status_check

**Comment by jimenez** [comment]
@mesosphere-mergebot ship_it_label_status_check

**Comment by jimenez** [comment]
@mesosphere-mergebot pr-status-checks

**Comment by orsenthil** [comment]
@jimenez  - the upstream.json in https://github.com/mesosphere/dcos-enterprise/pull/1982/files is not reflective of the HEAD of this repo. - a bump-ee is required to bring it up to date and reflect the tests from EE again.

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

**Comment by orsenthil** [comment]
@jimenez  - Actions to take

1) Please rebase / merge the changes of this branch to the latest in master.
2) do a `bump-ee` after that.

@lingmann  -  Please note - we have pulled the EE part in the train without the Open.

**Comment by spahl** [comment]
Signed off for merge

**Comment by lingmann** [comment]
Manually applying shipit to get on train after discussing with Seb and Isabel

**Pull request merged**